src/dosext/mfs/mfs.c

```diff
@@ -4294,6 +4299,47 @@ static int dos_fs_redirect(struct vm86_regs *state, char *stk)
       break;
     }
 
+    case GET_LARGE_DISK_SPACE: /* 0xa3 */
+    {
+      cds_t tcds = Addr(state, es, edi);
+      char *name = cds_current_path(tcds);
+      uint32_t avail, total;
```

Comment:
Do you think I should use 64bit types here and then while they are > 32bits halve them and double the blocksize?

Reply:
I think you should use 64bit types
to raise the size to maximum (2Tb)
is the 32bit value overflowed.